front/pages/api/v1/w/[wId]/data_sources/[dsId]/join-slack.ts

```diff
@@ -0,0 +1,165 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSourceViewFromVaultAndDataSourceId } from "@app/lib/api/data_source_view";
+import { getGlobalVault } from "@app/lib/api/vaults";
+import { withPublicAPIAuthentication } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+
+/**
```

Comment:
no swagger this is a private endpoint

Reply:
isn't it public?

Reply:
Not if behind system API key and no Open API spec.